src/Opdex.Platform.Infrastructure/Data/Handlers/Tokens/SelectTokensWithFilterQueryHandler.cs

```diff
@@ -215,24 +215,35 @@ private static string QueryBuilder(SelectTokensWithFilterQuery request)
 
         // re-sort back into requested order
         return PagingBackwardQuery.Replace(InnerQuery, query)
-            .Replace(OrderBySort, OrderByBuilder(request.Cursor.OrderBy,
-                                                 Enum.GetName(typeof(SortDirectionType), request.Cursor.SortDirection),
-                                                 reverse: true));
+                                  .Replace(OrderBySort, OuterOrderByBuilder(
+                                      request.Cursor.OrderBy, Enum.GetName(typeof(SortDirectionType),
+                                      request.Cursor.SortDirection)));
     }
 
-    private static string OrderByBuilder(TokenOrderByType cursorOrderBy, string direction, bool reverse)
+    private static string InnerOrderByBuilder(TokenOrderByType cursorOrderBy, string direction)
     {
-        var summaryPrefix = reverse ? "r" : "ts";
-        var tokenPrefix = reverse ? "r" : "t";
+        var tokenIdWithDirection = $"t.{nameof(TokenEntity.Id)} {direction}";
 
-        var tokenIdWithDirection = $"{tokenPrefix}.{nameof(TokenEntity.Id)} {direction}";
+        return cursorOrderBy switch
+        {
+            TokenOrderByType.Name => $" ORDER BY t.{nameof(TokenEntity.Name)} {direction}, {tokenIdWithDirection}",
+            TokenOrderByType.Symbol => $" ORDER BY t.{nameof(TokenEntity.Symbol)} {direction}, {tokenIdWithDirection}",
+            TokenOrderByType.PriceUsd => $" ORDER BY AVG(ts.{nameof(TokenSummaryEntity.PriceUsd)}) {direction}, {tokenIdWithDirection}",
+            TokenOrderByType.DailyPriceChangePercent => $" ORDER BY AVG(ts.{nameof(TokenSummaryEntity.DailyPriceChangePercent)}) {direction}, {tokenIdWithDirection}",
```

Comment:
Should this `AVG` be here?

Reply:
Yes the inner query has to order by the AVG; the outer query orders the already-averaged column returned from the inner query